src/Cli/Microsoft.TemplateEngine.Cli/Commands/install/BaseInstallCommand.cs

```diff
@@ -16,19 +21,88 @@ internal BaseInstallCommand(
             : base(hostBuilder, commandName, SymbolStrings.Command_Install_Description)
         {
             ParentCommand = parentCommand;
-            Arguments.Add(NameArgument);
+            Arguments.Add(PackageIdentifierArgument);
             Options.Add(InteractiveOption);
             Options.Add(AddSourceOption);
             Options.Add(ForceOption);
         }
 
-        internal static Argument<string[]> NameArgument { get; } = new("package")
+        internal static Argument<ITemplateIdentifierArgument[]> PackageIdentifierArgument { get; } = new("package")
         {
             Description = SymbolStrings.Command_Install_Argument_Package,
-            Arity = new ArgumentArity(1, 99)
+            Arity = ArgumentArity.OneOrMore,
+            CustomParser = ParseTemplateIdentifierArguments
         };
 
+        private static ITemplateIdentifierArgument[]? ParseTemplateIdentifierArguments(ArgumentResult result)
+        {
+            var templateIdentifierArguments = new List<ITemplateIdentifierArgument>();
+            foreach (var value in result.Tokens.Select(t => t.Value))
+            {
+                if (File.Exists(value))
+                {
+                    templateIdentifierArguments.Add(new FileBasedTemplateIdentifier(new FileInfo(value)));
+                }
+                else if (Directory.Exists(value) || Directory.Exists(Path.GetDirectoryName(value)))
+                {
+                    var environmentSettings = new EngineEnvironmentSettings(new Edge.DefaultTemplateEngineHost("dotnet", "1.0.0"), false, null, null, null, null);
+                    // expand the value pattern to allow for globbing, etc:
+                    foreach (var path in InstallRequestPathResolution.ExpandMaskedPath(value, environmentSettings))
+                    {
+                        templateIdentifierArguments.Add(new FileBasedTemplateIdentifier(new FileInfo(path)));
+                    }
+                }
+                else if (value.Contains("::") && ParsePackageIdentityWithVersionSeparator(value, "::") is PackageIdentity packageIdentity)
+                {
+                    Console.WriteLine(string.Format(LocalizableStrings.Colon_Separator_Deprecated, packageIdentity.Id, packageIdentity.Version is not null ? packageIdentity.Version : string.Empty));
+                    templateIdentifierArguments.Add(new PackageIdentifier(packageIdentity));
+                }
+                else if (ParsePackageIdentityWithVersionSeparator(value, "@") is PackageIdentity packageIdentityWithAt)
+                {
+                    templateIdentifierArguments.Add(new PackageIdentifier(packageIdentityWithAt));
+                }
```

Comment:
this is us pivoting across the different _variants_ - priority is important here to make sure we go 'most selective' to 'least selective'.

one interesting thing to call out is that we're 'expanding' the directory and wildcard scenarios to a list of packages to install directly here - this is a parsing-time operation and gives faster, actionable errors to the user